tests/test_code_gen.py

```diff
@@ -284,6 +284,13 @@ def test_comprehension(self):
         """
         self.assertAstEqual(source)
 
+        source = """
```

Comment:
We could also use some of the test cases in the original commit: https://hg.python.org/cpython/rev/cf91a929b81a#l4.6